### ISSUE TITLE
BasePath not generated with JAX-RS interfaces

### ISSUE THREAD

**Issue by jkerak** [zalando-stups/swagger-codegen-tooling#37]
I am currently generating JAX-RS annotated interfaces. I'd like to have all the swagger-generated enpoints have a base path, for example, of `/api`.  I tried adding `basePath: "/api"` to the root of my swagger.yaml, but it does not look like this is picked up during code generation (and I do not see anything referencing basePath in the mustache templates.  Is this something that can be done?

I found a related issue in swagger codegen: [1425](https://github.com/swagger-api/swagger-codegen/issues/1425)
Thanks

**Comment by LappleApple** [comment]
Hi @jkerak, thanks for the comment. @jbellmann, do you agree with the comment? If so, maybe @jkerak would be interested in helping with this.

**Comment by ender74** [comment]
This issue also exists for Spring annotated interfaces. I just checked with version 0.4.34 and the springinterfacesResponseEntityNoSwaggerAnnotations Template. The following swagger:

  "swagger": "2.0",
  "info": {
  "basePath": "/api",
  "paths": {
    "/dependencies": {
...
produces this Interface:

public interface DependenciesApi {
  @RequestMapping(value="/dependencies",method=RequestMethod.GET , produces={ "application/json" })
  ResponseEntity<DependenciesList> dependenciesGet( @RequestParam(value="names")
    List<String> names);
}

I think, the base Path should be added as Class Annotation:

@RequestMapping("api")
public interface DependenciesApi {
  @RequestMapping(value="/dependencies",method=RequestMethod.GET , produces={ "application/json" })
  ResponseEntity<DependenciesList> dependenciesGet( @RequestParam(value="names")
    List<String> names);
}

to make this work.

I compared it with swagger-codegen-maven-plugin (spring profile for version 2.2.1), but they don't generate the basePath either.
